docs/design/incrementaldownload.md

```diff
@@ -0,0 +1,65 @@
+# Incremental Download
+
+## Overview
+The feature is to allow Incremental Download of the Artifact in release so that all of the artifact items are not downloaded from the source provider and can be retrieved from a pre-maintained cache as well.
+
+## Task Changes
+Ideally we should be given the hashes of the files from the service itself but since services like Jenkins and Teamcity do not provide such hashes, we needed some generic way to get the hashes.
```

Comment:
Can you reword to:
If services storing builds provide hash of the artifact file content, it can be leveraged to perform incremental download of artifacts. Since most of the popular services (Jenkins/TeamCity) do not provide file content hashes, solution is to generate the hashes before publishing the artifact and use this information during artifact download.

Also move this to Overview section.